### PR TITLE
Storefront account update

### PR DESCRIPTION
**Pull request by v10110**

### CODE REVIEW

File: spec/requests/spree/api/v2/storefront/account_spec.rb [hunk 1]
```diff
@@ -39,4 +39,43 @@
     end
 
   end
+
+  describe 'account#update' do
+    let(:user) { create(:user) }
+    before { patch :update, user: user, params: params }
```
Comment (damianlegawiec) [inline spec/requests/spree/api/v2/storefront/account_spec.rb:45]:
```suggestion
    before { patch :update, params: params }
```

File: spec/requests/spree/api/v2/storefront/account_spec.rb [hunk 2]
```diff
@@ -39,4 +39,42 @@
     end
 
   end
+
+  describe 'account#update' do
```
Comment (damianlegawiec) [inline spec/requests/spree/api/v2/storefront/account_spec.rb:43]:
```suggestion
  describe 'account#update' do
    include_context 'API v2 tokens'
    
    let!(:user)  { create(:user_with_addresses) }
    let(:headers) { headers_bearer }
  
```

File: spec/requests/spree/api/v2/storefront/account_spec.rb [hunk 3]
```diff
@@ -39,4 +39,42 @@
...
+    before { patch :update, params: params }
```
Comment (damianlegawiec) [inline spec/requests/spree/api/v2/storefront/account_spec.rb:44]:
```suggestion
    before { patch '/api/v2/storefront/account', params: params, headers: headers }
```

File: lib/controllers/api/spree/v2/storefront/account_controller_decorator.rb [hunk 4]
```diff
@@ -10,6 +10,18 @@ def self.prepended(base)
           def create
             result = Spree::Account::Create.call(user_params: spree_user_params)
 
+            render_payload(result)
+          end
+
+          def update
+            result = Spree::Account::Update.call(user: spree_current_user, user_params: spree_user_params)
```
Comment (v10110) [inline lib/controllers/api/spree/v2/storefront/account_controller_decorator.rb:17]: @damianlegawiec should I use `spree_current_user` here? I need your confirm

Comment (damianlegawiec) [reply lib/controllers/api/spree/v2/storefront/account_controller_decorator.rb:17]: @v10110 yes, you need to use `spree_current_user`

**Pull request merged**